entity_api/src/coaching_session.rs

```diff
@@ -61,6 +64,465 @@ pub async fn delete(db: &impl ConnectionTrait, coaching_session_id: Id) -> Resul
+/// - The `validate()` method enforces these constraints at the API boundary
+///
+/// # Usage Example
+/// ```rust
+/// use entity_api::coaching_session::IncludeOptions;
+///
+/// // Create options requesting relationship and organization data
+/// let mut options = IncludeOptions::none();
+/// options.relationship = true;
+/// options.organization = true;
+/// options.validate().unwrap(); // Passes: organization depends on relationship
+///
+/// // This would fail validation:
+/// let mut invalid = IncludeOptions::none();
+/// invalid.organization = true;  // Without relationship: true
+/// assert!(invalid.validate().is_err()); // Error: organization requires relationship
+/// ```
+#[derive(Debug, Clone, Copy)]
+pub struct IncludeOptions {
+    pub relationship: bool,
+    pub organization: bool,
+    pub goal: bool,
+    pub agreements: bool,
+}
+
+impl IncludeOptions {
+    /// Creates an `IncludeOptions` with all resources disabled.
+    ///
+    /// This is the baseline configuration - only the base coaching session data
+    /// will be fetched without any related resources.
+    pub fn none() -> Self {
+        Self {
+            relationship: false,
+            organization: false,
+            goal: false,
+            agreements: false,
+        }
+    }
+
+    /// Returns true if any option requires loading coaching_relationships data.
+    ///
+    /// This helper method determines whether we need to execute the batch query
+    /// for coaching_relationships. Currently, both the `relationship` and `organization`
+    /// options require this data (organization is accessed via relationship.organization_id).
+    pub fn needs_relationships(&self) -> bool {
+        self.relationship || self.organization
+    }
+
+    /// Validates that the include options form a valid dependency graph.
+    ///
+    /// # Validation Rules
+    /// - `organization = true` requires `relationship = true`
+    ///   (organizations are accessed through coaching_relationships)
+    ///
+    /// # Errors
+    /// Returns `EntityApiErrorKind::InvalidQueryTerm` if validation fails.
+    ///
+    /// # Why This Matters
+    /// Early validation at the entity_api layer prevents invalid database queries
+    /// and provides clear error messages to clients about invalid include combinations.
+    pub fn validate(&self) -> Result<(), Error> {
+        // organization requires relationship (can't get org without relationship)
+        if self.organization && !self.relationship {
+            return Err(Error {
+                source: None,
+                error_kind: EntityApiErrorKind::InvalidQueryTerm,
+            });
+        }
+        Ok(())
+    }
+}
+
+/// Sort field for coaching sessions
+#[derive(Debug, Clone, Copy)]
+pub enum SortField {
+    Date,
+    CreatedAt,
+    UpdatedAt,
+}
+
+/// Sort order for queries
+#[derive(Debug, Clone, Copy)]
+pub enum SortOrder {
+    Asc,
+    Desc,
+}
```

Comment:
I believe these should belong in `web` as they are the concerns of incoming API requests. 

We already have a `SortOrder` defined [here](https://github.com/refactor-group/refactor-platform-rs/blob/main/web/src/params/sort.rs) and I think we should use that if we can.

If we have a good argument for moving this concept into `entity_api` then we will need to find a more generic solution to port the other domains that use `SortOrder` (in `web`) to the new pattern. 

Having two separate patterns, I think, might come back to bite us.

Reply:
Addressed in commit 77d70dc